### PR TITLE
Feature/4996 fork task data donation onboarding

### PR DESCRIPTION
**Pull request by SwissMountainFrog**
This PR is a FORK TASK from 4996 (based on [JIRA Ticket 4788](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-4788)) and [sub task 4999](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-4999).

IMG_0457
IMG_0458

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Scenes/App/AppOnboardingNavigationController.swift [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+		guard
+			let opacityDelegate = viewController as? NavigationBarOpacityDelegate,
+			let scrollView = viewController.scrollView
+		else {
+			return
+		}
+		// We can overwrite the existing observer, since Swift 4 block based observers clean themselves up.
+		scrollViewObserver = scrollView.observe(\.contentOffset) { [weak self] _, _ in
+			guard let self = self else { return }
+			guard viewController == self.topViewController else { return }
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Scenes/App/AppOnboardingNavigationController.swift:133]: Could be simplified:
```suggestion
			guard viewController == self?.topViewController else { return }
```

File: src/xcode/ENA/ENA/Source/Scenes/Onboarding/OnboardingInfoViewController.swift [hunk 2]
```diff
@@ -27,7 +27,7 @@ final class OnboardingInfoViewController: UIViewController {
 		self.store = store
 		self.client = client
 		self.supportedCountries = supportedCountries
-
+
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Scenes/Onboarding/OnboardingInfoViewController.swift:30]:
```suggestion
```

File: src/xcode/ENA/ENA/Source/Workers/Store/SecureStore.swift [hunk 3]
```diff
@@ -311,7 +311,7 @@ final class SecureStore: Store {
 	}
 
 	#endif
-
+
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Workers/Store/SecureStore.swift:314]:
```suggestion
```

**Review by pascal-brause — APPROVED** [review]

**Review by marcussc — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Scenes/App/AppOnboardingNavigationController.swift (hunk 1 above)
Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/Scenes/App/AppOnboardingNavigationController.swift:133]: 👍  But as this is a 1:1 copy of the AppNavigationController I'll keep this until the refactoring 😉

**Pull request merged**